BlueDBOnDisk/src/main/java/io/bluedb/disk/segment/Segment.java

```diff
@@ -28,10 +28,19 @@
 
 	private final FileManager fileManager;
 	private final Path segmentPath;
+	private final LockManager<Path> lockManager;
 
 	public Segment(Path segmentPath, FileManager fileManager) {
 		this.segmentPath = segmentPath;
 		this.fileManager = fileManager;
+		lockManager = fileManager.getLockManager();
+	}
+
+	// for testing only
+	protected Segment() {

```

Comment:
How do you feel about making a static method called `createTestSegment()` or something like that? Then you'd easily be able to make a test instance of the class without the risk of someone trying to use it in non-test environments.